openedx/adg/lms/applications/admin.py

```diff
@@ -68,7 +105,403 @@ class BusinessLineAdmin(admin.ModelAdmin):
+
+        fieldset_for_scores = (SCORES, {'fields': (PREREQUISITES, )})
+        fieldsets.append(fieldset_for_scores)
+
+        return tuple(fieldsets)
+
+    def _get_fieldset_for_resume_cover_letter(self, application):
+        """
+        Prepare and return fieldset for resume and cover letter provided with application.
+
+        Arguments:
+            application (UserApplication): Application under review
+
+        Returns:
+            tuple: Fieldset containing both fieldset title and a child tuple for fields.
+
+                Title for fieldset varies based on the provided data. If the applicant has provided:
+
+                    both cover letter and resume, title returned is 'RESUME & COVER LETTER',
+                    resume only, title returned is 'RESUME'
+                    cover letter only, either as an attachment or in text, title returned is 'COVER LETTER'
+
+                Fields returned as part of the fieldset also vary depending on the data that the applicant has provided
+        """
+        resume_cover_letter_file_fields = []
+
+        if application.cover_letter_and_resume:
+            fieldset_title = RESUME_AND_COVER_LETTER
+            resume_cover_letter_file_fields.append(RESUME)
+            if application.cover_letter_file:
+                resume_cover_letter_file_fields.append(COVER_LETTER_FILE)
+        elif application.resume:
+            fieldset_title = RESUME_ONLY
+            resume_cover_letter_file_fields.append(RESUME)
+        else:
+            fieldset_title = COVER_LETTER_ONLY
+            if application.cover_letter_file:
+                resume_cover_letter_file_fields.append(COVER_LETTER_FILE)
+
+        resume_cover_letter_display_fields = self._get_resume_cover_letter_display_fields(application)
+
+        resume_cover_letter_fields = resume_cover_letter_file_fields + resume_cover_letter_display_fields
+        fieldset = fieldset_title, {'fields': tuple(resume_cover_letter_fields)}
+
+        return fieldset
+
+    def _get_resume_cover_letter_display_fields(self, application):
+        """
+        Get display fields, if applicable, for resume and/or cover letter
+
+        Arguments:
+            application (UserApplication): Application under review
+
+        Returns:
+            list: Display fields
+        """
+        resume_cover_letter_display_fields = []
+
+        if application.resume:
+            if is_displayable_on_browser(application.resume):
+                resume_cover_letter_display_fields.append(RESUME_DISPLAY)
+
+        if application.cover_letter_file:
+            if is_displayable_on_browser(application.cover_letter_file):
+                resume_cover_letter_display_fields.append(COVER_LETTER_FILE_DISPLAY)
+        elif application.cover_letter:
+            resume_cover_letter_display_fields.append(COVER_LETTER_TEXT)
+
+        return resume_cover_letter_display_fields
+
+    readonly_fields = (
```

Comment:
Is there any specific reason to list these fields down here and not at the start?

Reply:
Wanted to group them with fieldsets, but have moved them to the start now.